contracts/CollectorProvider/CollectorProvider.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../LockProvider/LockDealProvider.sol";
+import "../ProviderInterface/IDeposit.sol";
+import "../ProviderInterface/IProviderSingleIdRegistrar.sol";
+
+contract CollectorProvider is IDeposit, IProviderSingleIdRegistrar {
+    LockDealProvider public lockProvider;
+
+    constructor(/*address _lockDealNFT,*/ address provider) {
+        require(
+            /*_lockDealNFT != address(0x0) &&*/ provider != address(0x0),
+            "invalid address"
+        );
+        //lockDealNFT = LockDealNFT(_lockDealNFT);
+        lockProvider = LockDealProvider(provider);
+    }
+
+    function deposit(uint256 poolId, uint256 amount) public override {
+        uint256[] memory params = lockProvider.getParams(poolId);
+        params[0] += amount;
+        params[1] = block.timestamp;
```

Comment:
we do not change the time. need to check that the block.timestamp is before the time.